spring/src/main/java/io/cloudevents/spring/webflux/CloudEventHttpMessageWriter.java

```diff
@@ -0,0 +1,121 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.cloudevents.spring.webflux;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import io.cloudevents.CloudEvent;
+import io.cloudevents.CloudEventData;
+import io.cloudevents.SpecVersion;
+import io.cloudevents.core.CloudEventUtils;
+import io.cloudevents.core.format.EventFormat;
+import io.cloudevents.core.message.MessageWriter;
+import io.cloudevents.rw.CloudEventRWException;
+import io.cloudevents.rw.CloudEventWriter;
+import io.cloudevents.spring.http.CloudEventsHeaders;
+import org.reactivestreams.Publisher;
+import reactor.core.publisher.Mono;
+
+import org.springframework.core.ResolvableType;
+import org.springframework.core.io.buffer.DataBuffer;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.http.ReactiveHttpOutputMessage;
+import org.springframework.http.codec.HttpMessageWriter;
+
+public class CloudEventHttpMessageWriter implements HttpMessageWriter<CloudEvent> {
+
+	@Override
+	public List<MediaType> getWritableMediaTypes() {
+		return Arrays.asList(MediaType.APPLICATION_OCTET_STREAM, MediaType.ALL);
+	}
+
+	@Override
+	public boolean canWrite(ResolvableType elementType, MediaType mediaType) {
+		return CloudEvent.class.isAssignableFrom(elementType.toClass());
+	}
+
+	@Override
+	public Mono<Void> write(Publisher<? extends CloudEvent> inputStream, ResolvableType elementType,
+			MediaType mediaType, ReactiveHttpOutputMessage message, Map<String, Object> hints) {
+		return Mono.from(inputStream).map(event -> CloudEventUtils.toReader(event))
+				.flatMap(reader -> reader.read(new ReactiveHttpMessageWriter(message)));
+	}
+
+	private static class ReactiveHttpMessageWriter
+			implements MessageWriter<CloudEventWriter<Mono<Void>>, Mono<Void>>, CloudEventWriter<Mono<Void>> {
+
+		private final ReactiveHttpOutputMessage response;
+
+		public ReactiveHttpMessageWriter(ReactiveHttpOutputMessage response) {
+			this.response = response;
+		}
+
+		// Binary visitor factory
+
+		@Override
+		public CloudEventWriter<Mono<Void>> create(SpecVersion version) {
+			this.response.getHeaders().set(CloudEventsHeaders.SPEC_VERSION, version.toString());
+			return this;
+		}
+
+		// Binary visitor
+
+		@Override
+		public ReactiveHttpMessageWriter withAttribute(String name, String value) throws CloudEventRWException {
+			this.response.getHeaders().set(CloudEventsHeaders.ATTRIBUTES_TO_HEADERS.get(name), value);
+			return this;
+		}
+
+		@Override
+		public ReactiveHttpMessageWriter withExtension(String name, String value) throws CloudEventRWException {
+			this.response.getHeaders().set("ce-" + name, value);
+			return this;
+		}
+
+		@Override
+		public Mono<Void> end(CloudEventData value) throws CloudEventRWException {
+			if (this.response.isCommitted()) {
```

Comment:
Is there some situation where this might happen? Also why you didn't added to the other `end()` and `setEvent()` too?

Reply:
I guess it might be over defensive. I removed this check (it was copied from the vertx example), and I'm pretty sure it will bubble up as a runtime exception if that condition even is met.